release/serve_tests/workloads/serve_handle_wide_ensemble.py

```diff
@@ -0,0 +1,185 @@
+"""
+This test is parity of
+release/serve_tests/workloads/deployment_graph_wide_ensemble.py
+Instead of using graph api, the test is using pure handle to
+test wide fanout graph.
+
+
+       -> Node_1
+      /          \
+INPUT --> Node_2  --> combine -> OUTPUT
+      \    ...   /
+       -> Node_10
+
+ 1) Intermediate blob size can be large / small
+ 2) Compute time each node can be long / short
+ 3) Init time can be long / short
+"""
+
+import time
+import asyncio
+import click
+
+from typing import List
+from typing import Optional
+
+import ray
+from ray import serve
+from ray.serve.handle import RayServeSyncHandle
+from ray.serve.context import get_global_client
+from serve_test_cluster_utils import (
+    setup_local_single_node_cluster,
+    setup_anyscale_cluster,
+)
+from serve_test_utils import save_test_results
+from benchmark_utils import benchmark_throughput_tps, benchmark_latency_ms
+
+DEFAULT_FANOUT_DEGREE = 4
+
+DEFAULT_NUM_REQUESTS_PER_CLIENT = 20  # request sent for latency test
+DEFAULT_NUM_CLIENTS = 1  # Clients concurrently sending request to deployment
+
+DEFAULT_THROUGHPUT_TRIAL_DURATION_SECS = 10
+
+
+@serve.deployment
+class Node:
+    def __init__(self, id: int, init_delay_secs=0):
+        time.sleep(init_delay_secs)
+        self.id = id
+
+    async def predict(self, input_data, compute_delay_secs=0):
+        await asyncio.sleep(compute_delay_secs)
+
+        return input_data + self.id
+
+
+@serve.deployment
+class CombineNode:
+    def __init__(self, input_nodes: List[RayServeSyncHandle], compute_delay_secs):
+        assert type(input_nodes) == list
+        print(type(input_nodes[0]))
+        self.input_nodes = input_nodes
+        self.compute_delay_secs = compute_delay_secs
+
+    async def predict(self, data):
+        results = [
+            node.predict.remote(data, self.compute_delay_secs)
+            for node in self.input_nodes
+        ]
+        results = await asyncio.gather(*results)
+        return sum(results)
+
+
+def construct_wide_fanout_graph_with_pure_handle(
+    fanout_degree, init_delay_secs=0, compute_delay_secs=0
+) -> RayServeSyncHandle:
+    nodes = []
+    for id in range(fanout_degree):
+        Node.options(name=str(id)).deploy(id, init_delay_secs=init_delay_secs)
+        nodes.append(get_global_client().get_handle(str(id), sync=True))
```

Comment:
can you also add a cli flag to configure use of sync vs. async deployment handle ? This will help to gather signals about how much gain we can get with async handle support in deployment graph.